### ISSUE TITLE
Subroutine interface method ignore arrays declarations

### ISSUE THREAD

**Issue by JoeffreyLegaux** [ecmwf-ifs/loki#59]
The interface method of Subroutine ignores arrays declarations.
This is due to the fact that it searches for declared symbols in the list of arguments names, however arrays symbols also include the arrays dimensions which are not part of the arguments.

```
            if any(v in arg_names for v in decl.symbols):
                assert all(v in arg_names and v.type.intent is not None for v in decl.symbols), \
                    "Declarations must have intents and dummy and local arguments cannot be mixed."
```                    

This can be fixed by searching for the symbols names instead of the whole symbols :

```
            if any(v.name in arg_names for v in decl.symbols):
                assert all(v.name in arg_names and v.type.intent is not None for v in decl.symbols), \
                    "Declarations must have intents and dummy and local arguments cannot be mixed."
```

**Comment by mlange05** [comment]
Great catch @JoeffreyLegaux , and many thanks for reporting this. The suggested fix has been submitted as PR #60 with additional test coverage.

**Comment by JoeffreyLegaux** [comment]
I did another small change, but I'm not sure if my usage is right and/or it is worth integrating.
I add cray pointers in my definitions, lines in the form of "POINTER(IP_MYVAR_, MYVAR)".
I insert them as Instrinsic nodes, so I search for those in the interface method to remove them :
```
        for decl in FindNodes((ir.VariableDeclaration, ir.ProcedureDeclaration)).visit(self.spec):
            if any(v.name in arg_names for v in decl.symbols):
                assert all(v.name in arg_names and v.type.intent is not None for v in decl.symbols), \
                    "Declarations must have intents and dummy and local arguments cannot be mixed."
                # Replicate declaration with re-scoped variables
                variables = as_tuple(v.clone(scope=routine) for v in decl.symbols)
                decl_map[decl] = decl.clone(symbols=variables)
            else:
                decl_map[decl] = None  # Remove local variable declarations
        for decl in FindNodes(ir.Intrinsic).visit(self.spec):
            if 'POINTER' in decl.text:
                decl_map[decl] = None # Remove Cray pointers declarations
        routine.spec = Transformer(decl_map).visit(self.spec)
```
